### PR TITLE
add warning for users who are not logged in and are on /edit or /solve

### PR DESCRIPTION
**Pull request by rupeshparab**
Ref - https://github.com/benetech/MathShare/issues/1073

### CODE REVIEW

**Comment by rupeshparab** [comment]
@jscholes 
I have resolved everything you have mentioned other than the 4th point.

@alexrcabral 
Is it fine if we remove the analytics event for clicking inside the input box for link?

File: src/components/Home/index.js [hunk 1]
```diff
@@ -282,6 +302,58 @@ class Home extends Component {
...
+                </div>
+                <div className="row">
+                    <div className={home.shareLink}>
+                        <label htmlFor="resumeWorkUrl" className="sROnly">
+                            {Locales.strings.work_link}
+                        </label>
+                        <input
+                            id="resumeWorkUrl"
+                            type="text"
+                            aria-labelledby="resumeWorkDesc"
```
Comment (jscholes) [inline src/components/Home/index.js:333]: `aria-labelledby` needs to be removed now that we have a `<label>`.

Comment (rupeshparab) [reply src/components/Home/index.js]: sorry had missed it, removed it now

**Comment by alexrcabral** [comment]
@jscholes To your fourth point, the intent of the analytics tracking is to have a means of determining whether the link is useful. For example, if people are using it, either through the copy button or through the next best measurement of highlighting the text in the text box to copy-paste manually, it would tell us that this is something that users need and use. 

If no one uses it, we could remove the warning + link or change it to something more useful. 

Are there alternative actions you would suggest tracking that would be more universal, to capture the event of users manually highlighting the link text to copy-paste external to Mathshare (e.g. to a note or email)?

**Comment by jscholes** [comment]
@alexrcabral I understand the intent, but I was more questioning the usefulness of knowing whether someone has just clicked on the field which may trigger some false positives.  It may be more useful to somehow detect that they've not only clicked, but interacted with it in a meaningful way (e.g. actually copied it).  Further, to have a unique link type for sharing, rather than it just being the same URL already being viewed.  That way, you can track how often people are actually using the links after they've been shared.

In any case, that's not strictly accessibility-related so I will proceed to review the latest changes independent of that issue.

**Comment by alexrcabral** [comment]
@jscholes Good point, and I think it's worth looking into in the future on how our link sharing is set up in order to get better insight into how people are interacting with the site.

**Pull request merged**